src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -163,8 +164,50 @@ private function collectOnClient(TraceableHttpClient $client): array
             unset($traces[$i]['info']); // break PHP reference used by TraceableHttpClient
             $traces[$i]['info'] = $this->cloneVar($info);
             $traces[$i]['options'] = $this->cloneVar($trace['options']);
+            $traces[$i]['curlCommand'] = $this->getCurlCommand($trace);
         }
 
         return [$errorCount, $traces];
     }
+
+    private function getCurlCommand(array $trace): string
+    {
+        $method = $trace['method'];
+        $options = $trace['options'];
+        $url = $trace['info']['url'];
+        
+        $command = ['curl'];
+
+        if ('GET' !== $method) {
+            $command[] = "-X$method";
+        }
+
+        foreach ($options as $option => $value) {
+            switch ($option) {
+                case 'auth_ntlm':
+                    $command[] = '--ntlm';
+                    // Not breaking here, the "-u" argument below is still valid.
+                case 'auth_basic':
+                    $command[] = '-u';
+                    $command[] = sprintf('"%s:%s"', $value[0], $value[1]);
+                    break;
+                case 'auth_bearer':
+                    $command[] = '--header';
+                    $command[] = sprintf('"Authorization: Bearer %s"', $value);
+                case 'headers':
+                    foreach ($value ?? [] as $headerName => $headerValue) {
+                        // Add `--header` argument for each header.
+                        // If the header value contains quotes, they will be escaped.
+                        $command[] = '--header';
+                        $command[] = sprintf('"%s: %s"', $headerName, str_replace('"', '\"', $headerValue));
```

Comment:
escaping only `"` is not enough. On Unix systems, double-quoted strings are interpolating variables.

Reply:
for reference, here is the code from the Firefox dev tools for the equivalent feature: https://searchfox.org/mozilla-central/source/devtools/client/shared/curl.js

Reply:
PHP native [`escapeshellarg()`](https://www.php.net/manual/fr/function.escapeshellarg.php) seems to do the job, so let's use it :)